### ISSUE TITLE
storage: leave intents behind after blind-writes experiencing write-too-old

### ISSUE THREAD

**Issue by jseldess** [cockroachdb/docs#6733]
PR: https://github.com/cockroachdb/cockroach/pull/44654

From release notes:
> An `UPDATE` returning a serialization failure error (code `40001`) now leaves behind a lock, helping the transaction succeed if it retries. This prevents starvation of transactions where an `UPDATE` is prone to conflicts. [#44654][#44654] {% comment %}doc{% endcomment %}

**Comment by rmloveland** [comment]
Hi @andreimatei !

tl;dr: I don't think this change needs docs, since it looks (to me) like an internal optimization.  Please shout if you disagree, though.

Longer version:

I have some questions/comments about this from a docs POV:

- Since we already document that client apps should retry on serialization / 40001 errors, it seems like this change is a good thing that should magically make their transactions work better

- However, it seems like this could be classed as an internal optimization of the `UPDATE` statement, and thus not necessarily something that needs to be on the `UPDATE` SQL statement page.

I'm inclined to think that the release note is sufficient documentation for this change, since it's an optimization.  And if we document it, we will have to change the docs again when/if a new/different locking optimization happens.

Therefore I'd like to close this issue without documenting this behavior of our locking, and just let users get this better experience automagically, without having to read/think about it. 

Do you agree?

**Comment by andreimatei** [comment]
Yeah, I don't think there's anything to document about this change specifically.

I think we'll need to start providing users with a more serious locking model than we have in the past, particularly because of the new `select for update` support. But that's separate.
